src/tests/integration/SparkLend.t.sol

```diff
@@ -0,0 +1,504 @@
+        vm.prank(admin); treasuryAdmin.transfer(address(treasury), address(adai), address(this), delta);
+        sparkPool.withdraw(address(dai), delta, address(adai));
+
+        assets = getTotalAssets(address(dai)) + 1;  // In case of rounding error we +1
+        liabilities = getTotalLiabilities(address(dai));
+        assertGe(assets, liabilities, "assets should be greater than or equal to liabilities");
+    }
+
+    function test_asset_liabilities_fix_full_utilization_flashloan() public {
+        uint256 assets = getTotalAssets(address(dai));
+        uint256 liabilities = getTotalLiabilities(address(dai));
+        if (assets >= liabilities) {
+            // Force the assets to become less than the liabilities
+            uint256 performanceBonus = daiInterestRateStrategy.performanceBonus();
+            vm.prank(admin); plan.file("buffer", performanceBonus * 4);
+            hub.exec(ilk);
+            sparkPool.borrow(address(dai), performanceBonus * 2, 2, 0, address(this));  // Supply rate should now be above 0% (we are over-allocating)
+
+            // Warp so we gaurantee there is new interest
+            vm.warp(block.timestamp + 365 days);
+            forceUpdateIndicies(address(dai));
+
+            assets = getTotalAssets(address(dai));
+            liabilities = getTotalLiabilities(address(dai));
+            assertLe(assets, liabilities, "assets should be less than or equal to liabilities");
+        }
+        
+        // Let's fix the accounting
+        uint256 delta = liabilities - assets;
+
+        // First trigger all spDAI owed to the treasury to be accrued
+        assertGt(getAccruedToTreasury(address(dai)), 0, "accrued to treasury should be greater than 0");
+        address[] memory toMint = new address[](1);
+        toMint[0] = address(dai);
+        sparkPool.mintToTreasury(toMint);
+        assertEq(getAccruedToTreasury(address(dai)), 0, "accrued to treasury should be 0");
+        assertGe(adai.balanceOf(address(treasury)), delta, "adai treasury should have more than the delta between liabilities and assets");
+
+        // Donate the excess liabilities back to the pool
+        // This will burn the liabilities while keeping the assets the same
+        vm.prank(admin); treasuryAdmin.transfer(address(treasury), address(adai), address(this), delta);
+
+        // Remove all DAI liquidity from the pool
+        uint256 poolBalance = dai.balanceOf(address(adai));
+        dai.setBalance(address(adai), 0);
+        dai.setBalance(address(this), 0);       // We have no DAI as well
+
+        // Withdrawing won't work because no available DAI
+        vm.expectRevert();
+        sparkPool.withdraw(address(dai), delta, address(adai));
+
+        // Flash loan to close out the liabilities
+        flashLender.flashLoan(this, address(dai), delta, "");
+
+        // Someone puts the DAI back into the pool
+        dai.setBalance(address(adai), dai.balanceOf(address(adai)) + poolBalance);
```

Comment:
Can we do it with actually pool deposit? it is part of the sequence we want to check.
Also we should use `delta` right?